### PR TITLE
SREP-1659: Extend machine API metrics w/ cluster info

### PR DESCRIPTION
**Pull request by lisa**
This also addresses SREP-1759 with the inclusion of the
`managed.openshift.io/exporter_source_code_hash` annotation which is
composed of `monitor/*` sha256'd.

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

### CODE REVIEW

**Review by dak1n1 — APPROVED** [review]
/lgtm

**Comment by dak1n1** [comment]
I tested this locally and it adds the hash like it's supposed to.

```
[dakini@nibbana managed-prometheus-exporter-machine-api ((c228b1e...) *)]$ git diff
diff --git a/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl b/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl
index 35e56f2..80da144 100644
--- a/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl
+++ b/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl
@@ -39,7 +39,7 @@ objects:
         template:
           metadata:
             annotations:
-              managed.openshift.io/exporter_source_code_hash: 3c3c7a2cb887b31b205c0032eaf655b03dba92e4d40a677c3d5e77440029056a
+              managed.openshift.io/exporter_source_code_hash: (stdin)= 3c3c7a2cb887b31b205c0032eaf655b03dba92e4d40a677c3d5e77440029056a
             labels:
```

**Comment by lisa** [comment]
huh, odd, it shouldn't add the `(stdin)` piece. I wonder if openssl is inconsistent across platforms!

File: project.mk [hunk 1]
```diff
@@ -6,4 +6,7 @@ GIT_ROOT?=$(shell git rev-parse --show-toplevel 2>&1)
 # WARNING: REPO_NAME will default to the current directory if there are no remotes
 REPO_NAME?=$(shell basename $$((git config --get-regex remote\.*\.url 2>/dev/null | cut -d ' ' -f2 || pwd) | head -n1 | sed 's|.git||g'))
 
-SELECTOR_SYNC_SET_DESTINATION?=${GIT_ROOT}/hack/00-osd-${REPO_NAME}.selectorsyncset.yaml.tmpl
\ No newline at end of file
+SELECTOR_SYNC_SET_DESTINATION?=${GIT_ROOT}/hack/00-osd-${REPO_NAME}.selectorsyncset.yaml.tmpl
+
+SOURCE_CODE_HASH ?= $(shell (cat monitor/* 2>/dev/null || echo "no-source-code") | openssl dgst -sha256)
```
Comment (rogbas) [inline project.mk]: Wouldn't it be the case of doing here: `openssl dgst -sha256 -binary | xxd -p -c 256`, to avoid the leading `(stdin)` text?

Comment (lisa) [reply project.mk]: @rogbas check the updated file, let me know what you think of the workaround?

**Comment by dak1n1** [comment]
The latest change works when I copy/paste it into my terminal like this:

```
[dakini@nibbana ~]$ (cat monitor/* 2>/dev/null || echo "no-source-code") | openssl dgst -sha256 | openssl dgst -sha256 | grep -o '[[:xdigit:]][[:xdigit:]]*$'
588bc6ee354d0a0e11a569345f2c950362d213a6ce185a85b7e37e1aebfcb522
```

But when I do a `make`, I'm still not getting the expected output:

```
[dakini@nibbana managed-prometheus-exporter-machine-api ((f14bade...))]$ git log -p
commit f14bade5ae8b54fa23b99ff21de82bc25ada105a (HEAD, lisa/srep-1659-extend-metrics)
Author: Lisa Seelye <lseelye@redhat.com>
Date:   Tue Jul 16 15:32:11 2019 -0400

    SREP-1659: Cross platform the checksum

    Signed-off-by: Lisa Seelye <lseelye@redhat.com>

diff --git a/project.mk b/project.mk
index b85ac1f..b2d7519 100644
--- a/project.mk
+++ b/project.mk
@@ -8,5 +8,8 @@ REPO_NAME?=$(shell basename $$((git config --get-regex remote\.*\.url 2>/dev/nul

 SELECTOR_SYNC_SET_DESTINATION?=${GIT_ROOT}/hack/00-osd-${REPO_NAME}.selectorsyncset.yaml.tmpl

-SOURCE_CODE_HASH ?= $(shell (cat monitor/* 2>/dev/null || echo "no-source-code") | openssl dgst -sha256)
+# Munge the output of the following because Linux openssl and MacOS openssl
+# somehow have different output formats MacOS merely prints the checksum
+# whereas Linux will print (stdin)= checksum.
+SOURCE_CODE_HASH ?= $(shell (cat monitor/* 2>/dev/null || echo "no-source-code") | openssl dgst -sha256 | openssl dgst -sha256 | grep -o '[[:xdigit:]][[:xdigit:]]*$')


commit c228b1edd8e6c1e88383471f6f2793fb937876b8
Author: Lisa Seelye <lseelye@redhat.com>
Date:   Tue Jul 16 13:13:25 2019 -0400

[dakini@nibbana managed-prometheus-exporter-machine-api ((f14bade...))]$ make
docker run --rm -v `pwd -P`:`pwd -P` python:2.7.15 /bin/sh -c "cd `pwd -P`; pip install pyyaml; scripts/generate_syncset.py -t scripts/templates/ -y deploy -d /home/dakini/go/src/github.com/openshift/managed-prometheus-exporter-machine-api/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl -r managed-prometheus-exporter-machine-api"
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting pyyaml
  Downloading https://files.pythonhosted.org/packages/a3/65/837fefac7475963d1eccf4aa684c23b95aa6c1d033a2c5965ccb11e22623/PyYAML-5.1.1.tar.gz (274kB)
Building wheels for collected packages: pyyaml
  Building wheel for pyyaml (setup.py): started
  Building wheel for pyyaml (setup.py): finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/16/27/a1/775c62ddea7bfa62324fd1f65847ed31c55dadb6051481ba3f
Successfully built pyyaml
Installing collected packages: pyyaml
Successfully installed pyyaml-5.1.1
You are using pip version 19.0.3, however version 19.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
scripts/generate_syncset.py:35: YAMLLoadWarning: calling yaml.load_all() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  return list(yaml.load_all(input_file))
[dakini@nibbana managed-prometheus-exporter-machine-api ((f14bade...) *)]$ git diff
diff --git a/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl b/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl
index 35e56f2..80da144 100644
--- a/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl
+++ b/hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl
@@ -39,7 +39,7 @@ objects:
         template:
           metadata:
             annotations:
-              managed.openshift.io/exporter_source_code_hash: 3c3c7a2cb887b31b205c0032eaf655b03dba92e4d40a677c3d5e77440029056a
+              managed.openshift.io/exporter_source_code_hash: (stdin)= 3c3c7a2cb887b31b205c0032eaf655b03dba92e4d40a677c3d5e77440029056a
             labels:
               name: sre-machine-api-status-exporter
             name: sre-machine-api-status-exporter
[dakini@nibbana managed-prometheus-exporter-machine-api ((f14bade...) *)]$
```

File: project.mk (hunk 1 above)
Comment (rogbas) [reply project.mk]: +1 for the filter with grep, but I didn't understand the double call to `openssl -dgst -sha256`

`openssl dgst -sha256 | openssl dgst -sha256 | grep -o '[[:xdigit:]][[:xdigit:]]*$'`

Comment (lisa) [reply project.mk]: the double call is a defect.

Comment (lisa) [reply project.mk]: Should be fixed now in 3d6fd0c

**Review by rogbas — APPROVED** [review]
/lgtm

**Comment by dak1n1** [comment]
It worked after I used `make -B` to make all build targets. Sorry about that!

**Comment by lisa** [comment]
> It worked after I used `make -B` to make all build targets. Sorry about that!

a4b4bcd should fix this in the future

**Pull request merged**